core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1565,6 +1647,39 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
         ErrorReport(StatusCodes.BadRequest, "may not grant readers compute access")
       )
     }
+    if (
+      workspace.workspaceType.equals(WorkspaceType.McWorkspace) &&
```

Comment:
It might be a bit cleaner to extract the common check for `WorkspaceType.McWorkspace` to a parent conditional rather than have it in all 3 conditionals here.